### PR TITLE
[HttpClient] Add `net.peer.name` and `net.peer.port` on metric dimensions

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Towards #3373 

## Changes

Added dimensions as per [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md#attributes)

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerMetricsDiagnosticListener.cs [hunk 1]
```diff
@@ -49,15 +49,33 @@ public override void OnEventWritten(string name, object payload)
...
+                    if (!request.RequestUri.IsDefaultPort)
+                    {
+                        tags = new TagList
+                        {
+                            new KeyValuePair<string, object>(SemanticConventions.AttributeHttpMethod, HttpTagHelper.GetNameForHttpMethod(request.Method)),
+                            new KeyValuePair<string, object>(SemanticConventions.AttributeHttpScheme, request.RequestUri.Scheme),
+                            new KeyValuePair<string, object>(SemanticConventions.AttributeHttpStatusCode, (int)response.StatusCode),
+                            new KeyValuePair<string, object>(SemanticConventions.AttributeHttpFlavor, HttpTagHelper.GetFlavorTagValueFromProtocolVersion(request.Version)),
+                            new KeyValuePair<string, object>(SemanticConventions.AttributeNetPeerName, request.RequestUri.Host),
+                            new KeyValuePair<string, object>(SemanticConventions.AttributeNetPeerPort, request.RequestUri.Port),
```
Comment (Kielek) [inline src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerMetricsDiagnosticListener.cs:65]: from the spec: `Conditionally Required: If not default (80 for http, 443 for https).`

IMO worth to check if it is really needed to add this attribute.

Comment (cijothomas) [reply src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerMetricsDiagnosticListener.cs:65]: this is already doing that right? port is only added if non-default.

Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerMetricsDiagnosticListener.cs:65]: @Kielek - I am checking if it is not a default port first. do you mean the port should not be added at all?

Comment (Kielek) [reply src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerMetricsDiagnosticListener.cs:65]: Sure... I have overlooked condition. Sorry!

File: src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerMetricsDiagnosticListener.cs [hunk 2]
```diff
@@ -49,15 +49,33 @@ public override void OnEventWritten(string name, object payload)
                 {
                     var request = response.RequestMessage;
 
+                    TagList tags;
+
                     // TODO: This is just a minimal set of attributes. See the spec for additional attributes:
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerMetricsDiagnosticListener.cs:54]: this todo can be removed now, right?

Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerMetricsDiagnosticListener.cs]: done

**Pull request merged**